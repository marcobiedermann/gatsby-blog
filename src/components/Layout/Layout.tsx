import dayjs from 'dayjs';
import { Link } from 'gatsby';
import React, { ReactNode } from 'react';
import Footer from '../Footer';
import Grid from '../Grid';
import Header from '../Header';
import Main from '../Main';
import * as styles from './Layout.module.css';

export interface LayoutProps {
  children: ReactNode;
}

function Layout(props: LayoutProps) {
  const { children } = props;

  return (
    <div className={styles.layout}>
      <Header>
        <Grid>
          <Link to="/">Gatsby Blog</Link>
        </Grid>
      </Header>
      <Main>
        <Grid>{children}</Grid>
      </Main>

      <Footer>
        <Grid>
          <p>
            Â© {dayjs().format('YYYY')}, Built with{' '}
            <a href="https://www.gatsbyjs.com" rel="noopener noreferrer" target="_blank">
              Gatsby
            </a>
          </p>
        </Grid>
      </Footer>
    </div>
  );
}

export default Layout;
