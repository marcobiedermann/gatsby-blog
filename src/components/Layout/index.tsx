import { Link } from 'gatsby';
import React from 'react';
import Footer from '../Footer';
import Grid from '../Grid';
import Header from '../Header';
import Main from '../Main';
import styles from './style.module.css';

const Layout = (props) => {
  const { children } = props;

  return (
    <div className={styles.layout}>
      <Header>
        <Grid>
          <Link to="/">Gatsby Blog</Link>
        </Grid>
      </Header>
      <Main>
        <Grid>{children}</Grid>
      </Main>

      <Footer>
        <Grid>
          Â© {new Date().getFullYear()}, Built with
          {` `}
          <a href="https://www.gatsbyjs.com">Gatsby</a>
        </Grid>
      </Footer>
    </div>
  );
};

export default Layout;
