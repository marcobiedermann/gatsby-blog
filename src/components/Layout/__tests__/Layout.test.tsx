import { render } from '@testing-library/react';
import React from 'react';
import Layout from '..';

beforeAll(() => {
  jest.useFakeTimers();
  jest.setSystemTime(new Date('2020-01-01'));
});

afterAll(() => {
  jest.useRealTimers();
});

describe('layout component', () => {
  it('should render correctly', () => {
    const { container } = render(<Layout>Content</Layout>);

    expect(container.firstChild).toMatchSnapshot();
  });
});
