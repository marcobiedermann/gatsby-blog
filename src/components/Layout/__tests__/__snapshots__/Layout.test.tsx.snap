// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`layout component should render correctly 1`] = `
<div
  class="layout"
>
  <header
    class="header"
  >
    <div
      class="grid"
    >
      <a
        href="/"
      >
        Gatsby Blog
      </a>
    </div>
  </header>
  <main
    class="main"
  >
    <div
      class="grid"
    >
      Content
    </div>
  </main>
  <footer
    class="footer"
  >
    <div
      class="grid"
    >
      <p>
        Â© 
        2020
        , Built with
         
        <a
          href="https://www.gatsbyjs.com"
          rel="noopener noreferrer"
          target="_blank"
        >
          Gatsby
        </a>
      </p>
    </div>
  </footer>
</div>
`;
